index.js

```diff
@@ -23,15 +25,42 @@ function createFunction(parameters, defaultType) {
             throw new Error('Unknown function type "' + type + '"');
         }
 
-        if (property === '$zoom') {
+        if (zoomAndFeatureDependent) {
+            var featureFunctions = {};
+            var featureFunctionStops = [];
+            for (var s = 0; s < parameters.stops.length; s++) {
+                var stop = parameters.stops[s];
+                if (featureFunctions[stop[0].zoom] === undefined) {
+                    featureFunctions[stop[0].zoom] = {
+                        type: parameters.type,
+                        property: parameters.property,
+                        stops: []
+                    };
+                }
+                featureFunctions[stop[0].zoom].stops.push([stop[0].value, stop[1]]);
+            }
+
+            for (var z in featureFunctions) {
+                featureFunctionStops.push([parseInt(z), createFunction(featureFunctions[z])]);
+            }
+            fun = function(zoom, feature) {
+                return evaluateCategoricalFunction({ stops: featureFunctionStops, base: parameters.base }, zoom)(zoom, feature);
```

Comment:
Since zoom+data paint attributes only evaluate data at exact stops there is no need for this. The interpolation happens in the shader. Interpolation across zooms will be needed for layout properties so I added it here: 1adb9b312cbd065bfd09656df7fc724397e19199